# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxPlatformFontList.cpp
# Commit: 06c26e973285
# Full Hash: 06c26e97328589418681f58d637df751451f04ca
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2020-05-26 21:37:52
# Regressor Bug: 1640119
# File Overlap Count: 1
# Description:
#   Bug 1640119 - Pass shared-memory blocks for the font list as part of SetXPCOMProcessAttributes. r=jwatt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D76507
# ==============================================================================

diff -r fbdbe0cdce92 -r 06c26e973285 gfx/thebes/gfxPlatformFontList.cpp
--- a/gfx/thebes/gfxPlatformFontList.cpp	Tue May 26 11:38:58 2020 +0000
+++ b/gfx/thebes/gfxPlatformFontList.cpp	Tue May 26 09:30:17 2020 +0000
@@ -2347,6 +2347,14 @@
   }
 }
 
+void gfxPlatformFontList::ShareFontListToProcess(
+    nsTArray<base::SharedMemoryHandle>* aBlocks, base::ProcessId aPid) {
+  auto list = SharedFontList();
+  if (list) {
+    list->ShareBlocksToProcess(aBlocks, aPid);
+  }
+}
+
 void gfxPlatformFontList::InitializeFamily(uint32_t aGeneration,
                                            uint32_t aFamilyIndex) {
   auto list = SharedFontList();